### PR TITLE
[CSL-2170] Remove MonadFail

### PR DESCRIPTION
**Pull request by int-index**

### CODE REVIEW

File: lib/test/Test/Pos/UtilSpec.hs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {-# LANGUAGE DataKinds    #-}
 {-# LANGUAGE RankNTypes   #-}
-{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE Pos #-}
```
Comment (neongreen) [inline lib/test/Test/Pos/UtilSpec.hs]: Wat

File: client/test/Test/Pos/Client/Txp/UtilSpec.hs [hunk 2]
```diff
@@ -316,8 +316,7 @@ getSignerFromList (HM.fromList . map swap . toList -> hm) =
 
 makeManyUtxoTo1Params :: InputSelectionPolicy -> Int -> Integer -> Integer -> Gen CreateMTxParams
 makeManyUtxoTo1Params inputSelectionPolicy numFrom amountEachFrom amountTo = do
-    [skFrom, skTo] <- nonrepeating 2
-
+    ~[skFrom, skTo] <- nonrepeating 2
```
Comment (avieth) [inline client/test/Test/Pos/Client/Txp/UtilSpec.hs]: Is this relevant to the rest of the patch?

Comment (int-index) [reply client/test/Test/Pos/Client/Txp/UtilSpec.hs]: Yes, this changes how pattern matching failure is desugared. With a lazy pattern it's not using `MonadFail`, in case of a mismatch the execution will continue, but each bound variable will be bottom (`skFrom` and `skTo` in this case).
